File: Frontend/Translator/src/integration/apiTranslate/domainObjects/responses/WSErrorResponse.js
/*
 * Licensed To: ThoughtExecution
 * Authored By: Rishi Raj Bansal
 * Developed in: December 2019
 *
*/

import domainObjectConstants from '../../constants/domainObjectConstants.js';
import FormBaseVO from '../../../../controller/formVOs/FormBaseVO';
import WSBaseResponse from './WSBaseResponse.js';


class WSErrorResponse extends WSBaseResponse{

    constructor(apiResponseObject) {
        super(apiResponseObject);
        this.code = apiResponseObject[domainObjectConstants.DO_RESPONSE_ERROR_CODE];
        this.error_message = apiResponseObject[domainObjectConstants.DO_RESPONSE_ERROR_MESSAGE];
        this.user_message = apiResponseObject[domainObjectConstants.DO_RESPONSE_USER_MESSAGE];
    }

    convertToFormVO(viewRequestObject){
        super.convertToFormVO(viewRequestObject);
        viewRequestObject[FormBaseVO.FORM_ATTR_SUBMIT_ERROR][FormBaseVO.PARAM_ERROR_FIELD_CODE] = this.code;
        viewRequestObject[FormBaseVO.FORM_ATTR_SUBMIT_ERROR][FormBaseVO.PARAM_ERROR_FIELD_ERROR_MESSAGE] = this.error_message;
        viewRequestObject[FormBaseVO.FORM_ATTR_SUBMIT_ERROR][FormBaseVO.PARAM_ERROR_FIELD_USER_MESSAGE] = this.user_message;

    }

}

export default WSErrorResponse;

File: Backend/Python/Translator/application/generic/domainObjects/responses/wsErrorResponse.py
from application.generic.domainObjects.responses.wsBaseResponse import *


class WSErrorResponse(WSBaseResponse):

    def __init__(self):
        super().__init__()
        self._code = None
        self._error_message = None
        self._user_message = None

    @property
    def code(self):
        return self._code

    @code.setter
    def code(self, code):
        self._code = code

    @property
    def error_message(self):
        return self._error_message

    @error_message.setter
    def error_message(self, error_message):
        self._error_message = error_message

    @property
    def user_message(self):
        return self._user_message

    @user_message.setter
    def user_message(self, user_message):
        self._user_message = user_message

    def save_data(self, validated_data):
        super().save_data(validated_data)
        self.code = validated_data.get('code')
        self.error_message = validated_data.get('error_message')
        self.user_message = validated_data.get('user_message')